validate_clerk_input.py

```diff
@@ -0,0 +1,22 @@
+# -*- coding: utf-8 -*-
+"""
+Created on Thu Aug 19 18:26:19 2021
+
+@author: vjham
+"""
+
+from product_list import products
+
+
+def validate_clerk_input(clerk_input):
+    
+    list_of_valid_product_ids = [str(product['id']) for product in products]
+    
+    if clerk_input in list_of_valid_product_ids:
+        return clerk_input
+    elif clerk_input.upper() == 'DONE':
+        return_value = clerk_input.upper()
+        return return_value
+    else:
+        amended_clerk_input = input("You entered an invalid input. Try again or enter 'DONE' to generate the final receipt: ")
```

Comment:
So weird - I could have sworn I had written another comment in here about unit testing...

Anyway. SO. this `validate_clerk_input` method is a great candidate for unit testing. however, as it is now, you will have to mock user input. 

I think right now, your `validate_clerk_input` method is doing too much. I am thinking it should just return a boolean (True or False) indicating whether the input is valid or not. The code that then handles what to do when the input is invalid should probably go somewhere else.